core/modules/system/system.module

```diff
@@ -3436,7 +3436,7 @@ function system_header_block($settings) {
  */
 function system_breadcrumb_block($settings = array()) {
   $crumbs = backdrop_get_breadcrumb();
-  if (isset($settings['current']) && $settings['current'] == TRUE) {
+  if (isset($settings['current']) && $settings['current'] == TRUE && !empty($crumbs)) {
```

Comment:
I would change the expression order, and first check the value of `$crumbs`.

```php
if (!empty($crumbs) && isset($settings['current']) && $settings['current'] == TRUE) {
```

PHP short-circuits Boolean expressions. If `$crumbs` is empty, it would not check the value of `$settings['current']`, and it would not execute the code in that `if`-block.